Add documentation for URIs in PEM files (#456)
latchset/pkcs11-provider: docs/provider-pkcs11.7.md

```PKCS11_PROVIDER_DEBUG=file:/dev/stderr,level:2```

USE IN OLDER APPLICATIONS (URIs in PEM files)
=============================================

It is strongly suggested to update applications to use the new
OSSL_STORE API provided by OpenSSL 3.0 which accepts URIs to
transparenly load keys from either files or any other supported

> mechanism including pkcs11 URIs.

Reviewer: would prefer to use uppercase PKCS#11 here and also in other places as it is still an abbreviation and term, rather than a word:
```suggestion
mechanism including PKCS #11 URIs.
```

> However, for those applications that cannot yet be changed there is

Reviewer: not sure about the english rules for the commas, but I feel like there should be one more separating the sub sentence:
```suggestion
However, for those applications that cannot yet be changed, there is
```

> support for generating a "wrapper" PEM file that contains the pkcs11
> URI needed to identify a key on the a token.

Reviewer: ```suggestion
tool to generate a "wrapper" PEM file that contains the PKCS #11 
URI needed to identify a key on the a token.
```

This PEM file can be loaded via the clasic methods used to parse
PEM/DER representations of keys and will trigger the use of the
pkcs11-provider decoders when the provider is loaded. An error will be

> returned if the provided is not pre-loaded or an older version of

Reviewer: ```suggestion
returned if the provider is not pre-loaded or an older version of
```
